src/main/java/io/gravitee/policy/basicauth/BasicAuthenticationPolicy.java

```diff
@@ -149,18 +151,25 @@ public void handle(Authentication authentication) {
         }
     }
 
-    private void sendAuthenticationFailure(Response response, PolicyChain policyChain) {
-        sendAuthenticationFailure(response, policyChain, "Unauthorized");
+    private void sendAuthenticationFailure(Request request, Response response, PolicyChain policyChain) {
+        sendAuthenticationFailure(request, response, policyChain, "Unauthorized");
     }
 
-    private void sendAuthenticationFailure(Response response, PolicyChain policyChain, String message) {
+    private void sendAuthenticationFailure(Request request, Response response, PolicyChain policyChain, String message) {
         String realmName = basicAuthenticationPolicyConfiguration.getRealm();
 
         if (realmName == null || realmName.trim().isEmpty()) {
             realmName = DEFAULT_REALM_NAME;
         }
 
         response.headers().set(HttpHeaderNames.WWW_AUTHENTICATE, "Basic realm=\"" + realmName + "\"");
-        policyChain.failWith(PolicyResult.failure(HttpStatusCode.UNAUTHORIZED_401, message));
+        policyChain.failWith(
+            PolicyResult.failure(
+                BASIC_AUTHENTICATION_UNAUTHORIZED,
+                HttpStatusCode.UNAUTHORIZED_401,
+                message,
+                Maps.<String, Object>builder().put("path", request.path()).put("method", request.method()).build()
```

Comment:
If you want to display path and method in your response template, you can use `{#request.path}` and `{#request.method}`.
I don't think we need to add those extra parameters.
Can we remove them ?